csr2transmart/mappers/ontology_mapper.py

```diff
@@ -0,0 +1,88 @@
+from typing import List, Dict, Sequence, Union
+
+from transmart_loader.transmart import Concept, TreeNode, ValueType, ConceptNode, TreeNodeMetadata, Value, TextValue
+
+from csr.csr import SubjectEntity, StudyEntity, Individual
+from csr2transmart.ontology_config import TreeNode as OntologyConfigTreeNode, \
+    ConceptNode as OntologyConfigConceptNode
+
+
+class OntologyMapper:
+    """
+    Map ontology config to tree nodes and concepts
+    """
+    def __init__(self, top_tree_node: str):
+        self.top_tree_node = top_tree_node
+        self.concept_code_to_concept: Dict[str, Concept] = {}
+
+    @staticmethod
+    def type_to_value_type(field_type) -> ValueType:
+        if field_type in ['integer', 'number']:
+            return ValueType.Numeric
+        if field_type in ['date']:
+            return ValueType.Date
+        else:
+            return ValueType.Categorical
+
+    @staticmethod
+    def entity_name_to_subject_dimension_value(entity_name: str) -> Value:
+        if entity_name == 'Individual':
+            return TextValue('patient')
+        else:
+            return TextValue(entity_name)
+
+    @staticmethod
+    def is_concept_node(node):
+        if isinstance(node, dict):
+            return node.get('concept_code') is not None
+        else:
+            return isinstance(node, OntologyConfigConceptNode) and node.concept_code is not None
+
+    def get_concept_type(self, entity_name: str, entity_field_name: str) -> ValueType:
+        entity_types = list(SubjectEntity.__args__)
+        entity_types.extend(list(StudyEntity.__args__))
+        entity_type: SubjectEntity = list(filter(lambda x: x.schema()['title'] == entity_name, entity_types))[0]
+        field_type = entity_type.schema()['properties'][entity_field_name]['type']
+        return self.type_to_value_type(field_type)
+
+    def map_concept_node(self, node: OntologyConfigConceptNode, path_elements: List[str]) -> ConceptNode:
+        entity_name, entity_field_name = node.concept_code.split('.')
+        path_elements.append(entity_field_name)
+        concept_path = '\\'.join(path_elements)
```

Comment:
This should perhaps be `'\\\\'` then, looking at what ends up in the generated files

Reply:
The `path_elements` list seems to be growing? E.g, `\Central Subject Registry\\01. Patient information\gender\birth_date\taxonomy\death_date\Informed_consent\ic_type\ic_version`...
We can also choose to have the concept path `'\\\\CSR\\\\' + node.concept_code`.  It does not have to be the same as the tree node path.

Reply:
Good idea, I changed this to the simplified version: `'\\\\CSR\\\\' + node.concept_code`